### ISSUE TITLE
use uniforms/bridge system to build document views

### ISSUE THREAD

**Issue by niklasdahlheimer** [vazco/uniforms#900]
Besides making forms to enter data, which I love to do with uniforms like this, 

<img src="https://user-images.githubusercontent.com/36920541/110933695-653a0400-832d-11eb-871e-7a737f7a44ae.JPG" height="400px">

I have to display the  same data as "read only views" at different parts of my app like this

<img src="https://user-images.githubusercontent.com/36920541/110933697-65d29a80-832d-11eb-9695-00c723b220d6.JPG" height="400px">

I would like to avoid building this views manually for different collections, because this is kind of boilerplate code and doesn't have nice features like automatic update of the component when the underlying schema changes . 
A workaround for me is to show the data through uniforms and disable the form. But that comes with the drawback that the data is not selectable and the visibility of diabled fields is bad because it's not meant to display data.

So I came around with a tought. This could be much more complicated than I expect now, but I would like to discuss this idea with you guys:

Uniform has everything it needs to visualize data on the base of an bridged schema. 
**Wouldn't it be great to have "view"-package (i.e. "uniforms-semantic-view") or a sort of  "viewform"-component (i.e. called"<AutoView>") which uses a bridged schema and a document-object as a model and creates a data view (as a list or a table) with it?**
Of course this would probably need "view-fields" also to provide the great uniforms workflow of autofields vs. dedicated fields

Probably I oversee something and this is already there, but I didn't find anything liek this for simple schema in npm/galaxy.

Greetings
Niklas

**Comment by wadamek65** [comment]
Hey there @niklasdahlheimer , thanks for the feature request!

As you've already mentioned, uniforms has everything it needs to work in a view-only mode. This is also a use-case that we have in some of our projects so your proposal & way of using uniforms makes complete sense.

What I'm afraid of is that in case of view-only mode too many variables come into play. When it comes to styling/views I feel like flexibility and configurability are the most important. Even an opinionated view-only theme needs these features to meet most of the use-cases. Adding more complexity on top of UI frameworks that are already very complex might not be a good idea and creating a new theme in uniforms is already easy enough to do in your own project.

But I still would like you to encourage you to play around on your own and maybe give us some sample implementation on how you would see this kind of package implemented. We would be happy to provide our feedback.

TL;DR: IMHO creating a view-only theme will either result in an over-engineered and hard to maintain package or not meet enough use-cases to be worth our while.

Maybe @radekmie or @Monteth have different views on this matter though.

**Comment by radekmie** [comment]
Hi @niklasdahlheimer. Have you tried the `readOnly` mode? It's _slightly_ less restrictive than the `disabled` one.

When it comes to view mode as is, I agree with @wadamek65 that it's out of the scope of this package. However, if you'd make such a package, we're here to help as I think it'd be a good addition. As a starting point, you could pick your theme and change fields into views one by one.
